src/Shared/Sdk/Providers/Services/PlatformBaseService.cs

```diff
@@ -0,0 +1,35 @@
+using ACMTTU.NoteSharing.Shared.SDK.Clients;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Storage.Blob;
+using System.Threading.Tasks;
+using System.Net.Http;
+
+namespace ACMTTU.NoteSharing.Shared.SDK.Services {
+    public abstract class PlatformBaseService {
+        protected CosmosClient dbClient;
+        protected CloudBlobClient storageClient;
+
+        public PlatformBaseService(IHttpClientFactory clientFactory) {
```

Comment:
Does this provide the basis for the dependency injection to work since the `PlatformBaseController` class also has this as it's only parameter? Otherwise the inheritance flow makes sense to me.

Reply:
‘PlatformBaseController’ and ‘Service’ basically do the same thing, but just in different places. Now u can connect to the database outside of the controller. I’m trying to figure out what you are having trouble understanding...

Reply:
I understand the inheritance stuff, what doesn't make sense to me is the dependency injection and how we get all of this stuff within the controller class as a parameter of the constructor. I think it is just me needing to learn about how dependency injection works to be honest. The code itself and all of the functionality it sets out to do is easy to understand.

Reply:
We should definitely sit down and i can show you a bit about dependency injection any why it is useful and how to properly use this class. @rkelley20 what time are you not busy on 10/30?